services/brig/src/Brig/IO/Intra.hs

```diff
@@ -288,8 +296,47 @@ notifyUserDeletionLocals ::
   List1 Event ->
   m ()
 notifyUserDeletionLocals deleted conn event = do
-  recipients <- list1 deleted <$> lookupContactList deleted
-  notify event deleted Push.RouteDirect conn (pure recipients)
+  luid <- qualifyLocal deleted
+  -- first we send a notification to the deleted user's devices
+  notify event deleted Push.RouteDirect conn (pure (list1 deleted []))
+  -- then to all their connections
+  connectionPages Nothing luid (toRange (Proxy @500))
+  where
+    handler :: [UserConnection] -> m ()
+    handler connections = do
+      -- sent event to connections that are accepted
+      case qUnqualified . ucTo <$> filter ((==) Accepted . ucStatus) connections of
+        x : xs -> notify event deleted Push.RouteDirect conn (pure (list1 x xs))
+        [] -> pure ()
+      -- also send a connection cancelled event to connections that are pending
+      d <- viewFederationDomain
+      forM_
+        (filter ((==) Sent . ucStatus) connections)
+        ( \uc -> do
+            now <- liftIO $ toUTCTimeMillis <$> getCurrentTime
+            -- because the connections are going to be removed from the database anyway when a user gets deleted
+            -- we don't need to save the updated connection state in the database
+            -- note that we switch from and to users so that the "other" user becomes the recipient of the event
+            let ucCancelled =
+                  UserConnection
+                    (qUnqualified (ucTo uc))
+                    (Qualified (ucFrom uc) d)
+                    Cancelled
+                    now
+                    (ucConvId uc)
+            let e = ConnectionUpdated ucCancelled Nothing Nothing
+            onConnectionEvent deleted conn e
```

Comment:
is there a reason the local and the remote part look so different?  (just curious)